src/TaskRunner/Commands/BuildCommands.php

```diff
@@ -326,66 +326,88 @@ public function buildAssets(ConsoleIO $io, array $options = [
-                if (!in_array($taskRunnerConfigFile, $files)) {
-                    $dir = Toolkit::getToolkitRoot() . '/resources/assets';
-                    $collection->taskExecStack()
-                        ->exec("cp $dir/$taskRunnerConfigFile $theme_dir/$taskRunnerConfigFile")
-                        ->stopOnFail();
-                }
+        // Run and return task collection.
+        return $collection->run();
+    }
 
+    /**
+     * Runs the toolkit:build-assets on execute mode.
+     */
+    private function buildAssetsInstall($collection, $theme_dir, $allowedTaskRunners, $taskRunners, $files, $options)
+    {
+        $collection->taskExecStack()
+            ->dir($theme_dir)
+            ->exec('apt-get update')
+            ->exec('apt-get install ruby-sass -y')
+            ->exec('npm init -y')
+            ->stopOnFail();
+
+        // Check if 'theme-task-runner' file exists.
+        // Create a new one from source if doesn't exist.
+        foreach ($allowedTaskRunners as $allowedTaskRunner => $configFile) {
+            if (in_array($allowedTaskRunner, $taskRunners) && !in_array($configFile, $files)) {
+                $dir = Toolkit::getToolkitRoot() . '/resources/assets';
                 $collection->taskExecStack()
-                    ->dir($theme_dir)
-                    ->exec('npm init -y --scope')
-                    ->exec("npm install {$options['build-npm-packages']} --save-dev")
-                    ->exec('./node_modules/.bin/' . $options['theme-task-runner'])
+                    ->exec("cp $dir/$configFile $theme_dir/$configFile")
                     ->stopOnFail();
-
-                // Run and return task collection.
-                return $collection->run();
             }
-        } else {
-            $this->say("The theme '{$options['default-theme']}' couldn't be found on the '{$options['custom-code-folder']}' folder.");
+        }
+        $collection->taskExecStack()
+            ->dir($theme_dir)
+            ->exec("npm install {$options['build-npm-packages']} --save-dev")
+            ->stopOnFail();
+    }
+
+    /**
+     * Runs the toolkit:build-assets on execute mode.
+     */
+    private function buildAssetsExecute($allowedTaskRunners, $options, $files, $collection, $theme_dir)
+    {
+        if (!in_array($allowedTaskRunners[$options['execute']], $files)) {
+            $this->say('Config file not found! Make sure you have run the target toolkit:build-assets before use the execute mode.');
             return 0;
         }
+        if ($options['ecl-command']) {
+            $collection->taskExecStack()
+                ->dir($theme_dir)
+                ->exec($this->getNodeBin($options['execute']) . " {$options['ecl-command']}")
+                ->stopOnFail();
+        } else {
+            $collection->taskExecStack()
+                ->dir($theme_dir)
+                ->exec($this->getNodeBin($options['execute']))
+                ->stopOnFail();
+        }
```

Comment:
```suggestion
        $command = $this->getNodeBin($options['execute']);
        if ($options['ecl-command']) {
            $command .= ' ' . $options['ecl-command'];
        }
        $collection->taskExecStack()->dir($theme_dir)->exec($command)->stopOnFail();
```

@zarabatana what do you think about simplify this condition?